src/lib/consensus/proof_of_stake.ml

```diff
@@ -1397,29 +1343,27 @@ module Data = struct
     let length = Constants.Checkpoint_window.per_year
 
     module Repr = struct
+      [%%versioned
       module Stable = struct
         module V1 = struct
-          module T = struct
-            type t =
-              { (* TODO: Make a nice way to force this to have bounded (or fixed) size for
+          type t =
+            { (* TODO: Make a nice way to force this to have bounded (or fixed) size for
                    bin_io reasons *)
-                prefix:
-                  Coda_base.State_hash.Stable.V1.t Core.Fqueue.Stable.V1.t
-              ; tail: Hash.Stable.V1.t }
-            [@@deriving sexp, bin_io, compare, version {unnumbered}]
-
-            let digest ({prefix; tail} : t) =
-              let rec go acc p =
-                match Fqueue.dequeue p with
-                | None ->
-                    acc
-                | Some (h, p) ->
-                    go (merge h acc) p
-              in
-              go tail prefix
-          end
+              prefix: Coda_base.State_hash.Stable.V1.t Core.Fqueue.Stable.V1.t
+            ; tail: Hash.Stable.V1.t }
+          [@@deriving sexp, compare]
 
-          include T
+          let to_latest = Fn.id
+
+          let digest ({prefix; tail} : t) =
```

Comment:
It would be good to lift `digest` out of the version-specific module here if we can.

Reply:
At this line: 

  https://github.com/CodaProtocol/coda/blob/4d6e9b15098ae8f02ed5518585707f87834a7b2d/src/lib/consensus/proof_of_stake.ml#L1402
  
we're digesting a value from a particular `Repr` version.

Reply:
Can I recommend we move that out of the versioned module too? Doesn't seem like it needs to have anything to do with the serialised version..